sklearn/inspection/tests/test_partial_dependence.py

```diff
@@ -397,10 +403,9 @@ def test_partial_dependence_sample_weight():
     assert np.corrcoef(pdp, values)[0, 1] > 0.99
 
 
-@if_matplotlib
 def test_plot_partial_dependence():
     # Test partial dependence plot function.
-    import matplotlib.pyplot as plt  # noqa
+    plt = pytest.importorskip('matplotlib.pyplot')
```

Comment:
IMO having a decorator makes it more obvious that the test is matplotlib related.

Also, it is strange to use this pattern in some places while the `skip_if_no_matplotlib` decorator is used in some others

Reply:
I'm split. it was the suggestion of @ogrisel IRL. His point was to say that we have a decorator and then a lazily import while now you have only the import which skip the test if required.

I'll like the `mark.skipif` since it can be used in different setups.

Reply:
why not just avoid the decorator and always use `pytest.importorskip('matplotlib.pyplot')` then?

I'd rather keep things consistent.

But I'm OK too with decorator + lazy import.

Reply:
I find the idiom:

```python
     plt = pytest.importorskip('matplotlib.pyplot') 
```

good enough. I don't think we need a decorator for that.